arm/Microsoft.EventHub/namespaces/deploy.bicep

```diff
@@ -179,12 +187,6 @@ resource eventHubNamespace 'Microsoft.EventHub/namespaces@2021-06-01-preview' =
     zoneRedundant: zoneRedundant
     isAutoInflateEnabled: isAutoInflateEnabled
     maximumThroughputUnits: maximumThroughputUnits_var
-    networkAcls: !empty(networkAcls) ? {
```

Comment:
Out of curiosity - is this one of those objects where we have both a property & child-resource - and deploying the namespace without the property 'wipes' the settings?

This could be tested by deploying the resource with everything and then once without the child so see if the networkAcls remain

Reply:
Given that we are using a set of 'defaults', the `networkRuleSet` will always get deployed, hence if someone adds configuration via the parameter file, which is obviously overwritten, but if they remove the parameter and then run again, then it will go back to the default parameter value.

For the scenario above: And If we remove the 'defaults'. What happens is that lets say you configure the EH without `networkRuleSets`, and then you deploy it again and this time add `networkRuleSets` to the parameter file with specific config. The deployment goes in and applies those new configs. Removing that parameter and doing the deployment again **"Doesn't"** remove the previous settings away (unlike our friendly subnets/routetables/nsgs).

So if we do not want to override, we need to remove the 'defaults', and set an `if` condition to not trigger the `networkRuleSets`. This, however, doesn't allow enabling 'secure by default', as the default behavior from Azure is to keep it public by default whenever you deploy a vanilla EventHub namespace.

Reply:
I wouldn't consider what the portal does as secure by default. I'd here suggest not to enforce a default especially if the end result on a vanilla resource would be the same as just not enabling the child. That would be primarily to keep idempotency as per @MrMCake comment and also for consistency with the approach of not deploying child resources by default.

Reply:
Removed the defaults, and set a condition on the networkRuleSets module